File: 0513-Find Bottom Left Tree Value/0513-Find Bottom Left Tree Value.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int findBottomLeftValue(TreeNode* root) {
        queue<TreeNode*> Q;
        Q.push(root);
        int result;
        while (!Q.empty())
        {
            int qs = Q.size();
            result = Q.front()->val;
            for (int i = 0; i < qs; ++i)
            {
                TreeNode* node = Q.front();
                Q.pop();
                if (node->left)
                {
                    Q.push(node->left);
                }
                
                if (node->right)
                {
                    Q.push(node->right);
                }
            }
        }
        
        return result;
    }
};

File: 0501-0600/0513-Find Bottom Left Tree Value/0513-Find Bottom Left Tree Value.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

import collections
class Solution:
    def findBottomLeftValue(self, root: TreeNode) -> int:
        Q = collections.deque([root])
        while Q:
            result = Q[0].val
            l = len(Q)
            for i in range(l):
                node = Q.popleft()
                if node.left:
                    Q.append(node.left)
                if node.right:
                    Q.append(node.right)
        return result
